src/Address.cpp

```diff
@@ -0,0 +1,37 @@
+// Copyright © 2017-2022 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "Address.h"
+#include "Coin.h"
+
+namespace TW {
+
+Address::Address(const std::string& address, enum TWCoinType coin) {
+    auto normalized = TW::normalizeAddress(coin, address);
+    if (normalized.empty()) {
+        throw std::invalid_argument("unable to create address from string, not normalized");
```

Comment:
better to use factory method that able to return null value (same behavior like before)